[CCR] Polish copy and layout of Auto-follow Pattern Form

cjcenizal opened the pull request:
I got some feedback from the design team on the layout of the prefix and suffix fields. I also took a swing at improving the copy.

![image](https://user-images.githubusercontent.com/1238659/49492665-0f04bf00-f80e-11e8-8409-3a7c41c32566.png)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2438/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2730/)

yaronp68 commented:
Save button is active when required form fields are empty. Preferably button should be disabled at this state

![image](https://user-images.githubusercontent.com/1367657/49649474-32ec1e80-fa21-11e8-8b1c-b6c79eabf1a4.png)

cjcenizal commented:
@yaronp68 The validation messages only appear when the user hits the "Save" button. If we disable it initially then that means we'd have to run validation as you type to decide when to enable the "Save" button. If we show validation as you type then you'll see error messages before you're done entering input, which is a poor UX. If we _don't_ show validation as you type then the "Save" button will be disabled if you've entered invalid input but you won't know why.

cjcenizal commented:
I updated the copy per feedback, added help text, and improved the prefix and suffix validation rules.

![image](https://user-images.githubusercontent.com/1238659/49662412-9d409700-fa00-11e8-98de-9978f5d235e8.png)

![image](https://user-images.githubusercontent.com/1238659/49662419-a03b8780-fa00-11e8-84eb-7048b768757b.png)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2797/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2823/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/2840/)
